scrunch/datasets.py

```diff
@@ -141,23 +141,30 @@ def _get_dataset(dataset, connection=None, editor=False, project=None):
         else:
             shoji_ds = project.get_dataset(dataset).resource
```

Comment:
If `project.get_dataset()` returns the `Dataset` instance, why you're fetching its `.resource` again and making another `Dataset` instance?

Reply:
It was like that before my intervention with this PR, all I can guess is that `scrunch.datasets.get_dataset(...)` expects the shoji document, and not the Dataset instance.

I also see that `scrunch.datasets.get_dataset(...)` gonna be deprecated, so perhaps we can make it behave better on the newer methods.

Reply:
Ok, it's fine to do it in another ticket.